scripts/check-container-environment.sh

```diff
@@ -11,7 +11,7 @@
 
 set -euo pipefail
 
-KIND_VERSION="v0.14.0"
+KIND_VERSION="v0.20.0"
```

Comment:
I wonder if we should pin that version here as you see we forgot to update it. The best approach would be to pull it from `.github/test_dependencies.yaml` (but it doesn't define kind version for now) or have it pinned in `third_party/go.mod`.

But on the other hand since kind is available on GH hosted runner and we rather expect developers to have it installed on local environment too we can get rid entirely of this fallback for automatic installation of kind and not to pin the version here. WDYT?

Reply:
Yeah, given that `KinD` is already installed on GH runners, I think we could drop the version pin in this script. BTW maybe we should make this script simply do what its name suggests and just check if `KinD` and `docker` are installed, but not install them if not found.

Reply:
I think it's ok, I will remove the install logic.

Reply:
In order to unblock @pangruoran we can bump the version here and create a follow up PR so to get the version from `.github/test_dependencies.yaml`.

How does that sound to you @czeslavo, @programmer04 ?

Reply:
Yup we can do it, but I think this version wouldn't have much meaning because in GH runners we use that kind which GH ships, so it may be a surprise for someone

Reply:
So maybe we can merge it first, and if there is user feedback or other issues later on, we can decide whether to include the version of "kind" as a dependency.

Reply:
Let's merge this @pangruoran to unblock you. I suggest to create a follow up issue to discuss alternatives (we shouldn't have the version hardcoded in the script like that. @programmer04 is right that on CI `kind` is preinstalled but [the makefile target](https://github.com/Kong/kubernetes-ingress-controller/blob/9b81665dd679539b8ce4298ea4b3cad5aec5b00c/Makefile#L409-L411) that uses this script is used in other places (other makefiles targets) and it would be nice to have this defined in 1 place.